### PR TITLE
Add  deploy checklist / update deploy scripts

### PR DESCRIPTION
**Pull request by codenamejason**
Changes Included:
1.  Adds updated deploy scripts.
2. Adds deploy all networks with one command (make sure all wallets are funded or this will fail).
3. Adds gas reports.
4. Adds snapshots.
5. Adds updated to use bun instead of yarn --> happy to chat about this if anyone is not pleased about it but bun is way faster.
6. Adds all fixes for audit from another PR.
7. Adds verification.

### CODE REVIEW

**Comment by thelostone-mc** [comment]
Let's figure out a neater way as opposed to documenting every chain and all the steps.
Ideally we'd want something detailed which we can refer for every chain . Kinda like a deploy template (with deploy commands so that it's useful )

https://github.com/allo-protocol/allo-contracts/blob/v1.0.0/docs/DEPLOY_STEPS.md

**Comment by codenamejason** [comment]
I have updated to bun for running command and managing the packages, it is sooo much faster you wont believe it. Here is the install docs to make sure you have globally on your machine --> https://bun.sh/docs/installation

@thelostone-mc @0xKurt

File: hardhat.config.ts [hunk 1]
```diff
@@ -277,41 +370,41 @@ export default config;
...
-    const outputPath = path.resolve(__dirname, "DEPLOYMENT_INFORMATION.md");
-    fs.appendFileSync(outputPath, mdContent.trim());
+//     // Write the formatted information to a Markdown file
+//     const outputPath = path.resolve(__dirname, "DEPLOYMENT_INFORMATION.md");
+//     fs.appendFileSync(outputPath, mdContent.trim());
 
-    console.log(`Information has been written to ${outputPath}`);
-  });
+//     console.log(`Information has been written to ${outputPath}`);
+//   });
```
Comment (thelostone-mc) [inline hardhat.config.ts:410]: ^ why are we commenting this out

File: package.json [hunk 2]
```diff
@@ -56,8 +59,8 @@
     "eslint-plugin-no-only-tests": "^3.1.0",
     "eslint-plugin-node": "^11.1.0",
     "ethereum-waffle": "^4.0.10",
-    "ethers": "^6.8.0",
-    "hardhat": "^2.18.2",
+    "ethers": "^6.7.1",
+    "hardhat": "^2.17.4",
```
Comment (thelostone-mc) [inline package.json:63]: ^ Do we need to downgrade ?

File: scripts/core/deployContractFactory.ts [hunk 3]
```diff
@@ -47,10 +48,7 @@ export async function deployContractFactory() {
   };
 
   deploymentIo.write(objToWrite);
-
   await delay(20000);
-  await verifyContract(instance.target.toString(), []);
```
Comment (thelostone-mc) [inline scripts/core/deployContractFactory.ts:52]: Shouldn't we keep the verifying?

File: foundry.toml [hunk 4]
```diff
@@ -7,9 +7,19 @@ src = 'contracts'
...
 remappings = []
 
+# Gas reporting
+gas_reports = ["ContractFactory", "Registry", "Allo"]
+gas_reports_ignore = [
+  "/test/utils/*",
+  # "TestStrategy",
+  # "QVBaseStrategyTestMock",
+  # "Permit2",
+  # "MockStrategy",
```
Comment (thelostone-mc) [inline foundry.toml:20]: ^ Shouldn't these be uncommented

File: package.json (hunk 2 above)
Comment (codenamejason) [reply package.json:63]: This was left from zk deployment. Will update.

File: hardhat.config.ts (hunk 1 above)
Comment (codenamejason) [reply hardhat.config.ts:410]: it's not working and will be removed

File: package.json (hunk 2 above)
Comment (thelostone-mc) [reply package.json:63]: @0xKurt just calling this out

**Review by thelostone-mc — APPROVED** [review]

**Review by 0xKurt — APPROVED** [review]

**Pull request merged**